applications/StructuralMechanicsApplication/python_scripts/structural_response.py

```diff
@@ -135,6 +149,8 @@ def __init__(self, identifier, response_settings, model):
 
         self.primal_analysis = StructuralMechanicsAnalysis(model, ProjectParametersPrimal)
         self.primal_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.SHAPE_SENSITIVITY)
+        self.primal_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.YOUNGS_MODULUS_SENSITIVITY)
+
```

Comment:
I think this EigenFrequencyResponseFunction,  StrainEnergyResponseFunction (and MassResponsefunction?) are only implemented for shape sensitivities at the moment.

Note: there is a difference in the implementation of these first response functions and the "AdjointResponseFunction" below, that has a more general implementation that also allows element properties to be used as variables.

Reply:
I can confirm that these routines return dm/dE, dm/dt, dU/dE, dU/dt = 0. Removed and added error.

Reply:
well obviously a couple should be zero, but those that are obvioiusly neq 0 are zero.

Reply:
Explanation: `EigenFrequencyResponseFunction, StrainEnergyResponseFunction, MassResponsefunction` have been the first implementations of response functions in Kratos, specifically with application in ShapeOptimization in mind. 
All other response functions with "adjoint" in there name, have been implemented by @MFusseder following a much more general structure that allows computing gradients w.r.t different variables than shape only.